src/app/games/gipf/tests/GipfPiece.spec.ts

```diff
@@ -1,19 +1,12 @@
+import { NumberEncoderTestUtils } from 'src/app/jscaip/tests/Encoder.spec';
 import { GipfPiece } from '../GipfPiece';
 
 describe('GipfPiece', () => {
-    function checkEncodeDecode(piece: GipfPiece): void {
-        expect(GipfPiece.encoder.decode(GipfPiece.encoder.encode(piece))).toBe(piece);
-    }
     it('should correctly encode and decode empty piece', () => {
-        checkEncodeDecode(GipfPiece.EMPTY);
+        NumberEncoderTestUtils.expectToBeCorrect(GipfPiece.encoder, GipfPiece.EMPTY);
     });
     it('should correctly encode and decode player pieces', () => {
-        checkEncodeDecode(GipfPiece.PLAYER_ZERO);
-        checkEncodeDecode(GipfPiece.PLAYER_ONE);
-    });
-    it('should never encode above the maxValue', () => {
-        expect(GipfPiece.encoder.encode(GipfPiece.EMPTY) <= GipfPiece.encoder.maxValue()).toBeTrue();
-        expect(GipfPiece.encoder.encode(GipfPiece.PLAYER_ZERO) <= GipfPiece.encoder.maxValue()).toBeTrue();
-        expect(GipfPiece.encoder.encode(GipfPiece.PLAYER_ONE) <= GipfPiece.encoder.maxValue()).toBeTrue();
+        NumberEncoderTestUtils.expectToBeCorrect(GipfPiece.encoder, GipfPiece.PLAYER_ZERO);
+        NumberEncoderTestUtils.expectToBeCorrect(GipfPiece.encoder, GipfPiece.PLAYER_ONE);
```

Comment:
Why not Player ?

Reply:
Player plutôt que GipfPiece tu veux dire ? Si oui, oui ça pourrait mais c'est pas le sujet de cette PR donc ticketiser ?

Reply:
Certes! Surtout qu'idealement Gipf ce serait FourStatePiece vu qu'y a les None qui rendraient les tests vachement mieux (voir l'hexagone dans le carré et tout)

Reply:
D'ailleurs le FourStatePiece, à part rendre les tests visibles, est-ce que ça rajoute pas de la complexité au code pour aucune autre utilité du point de vue du code ? (Il y a un type de pièce qui est sur le plateau mais qui est impossible de manipuler vu que ça correspond aux endroits non existants; du coup si tu switch sur un FourStatePiece, t'as à chaque fois une branche pour ce cas non couvrable ?)